swarm/pss/pss.go

```diff
@@ -531,10 +557,11 @@ func (self *Pss) SendAsym(pubkeyid string, topic whisper.TopicType, msg []byte)
 	//pubkey := self.pubKeyIndex[pubkeyid]
 	pubkey := crypto.ToECDSAPub(common.FromHex(pubkeyid))
 	if pubkey == nil {
-		return errors.New(fmt.Sprintf("Invalid public key id %x", pubkey))
+		return fmt.Errorf("Invalid public key id %x", pubkey)
 	}
 	psp := self.pubKeyPool[pubkeyid][topic]
```

Comment:
We need a check here for `psp` being valid - I have been sending ill-formatted JSON-RPC, and it would crash go in `self.send(*psp.address.....)`